File: app/src/main/java/com/travlog/android/apps/services/apirequests/SignUpBody.java
package com.travlog.android.apps.services.apirequests;

import android.os.Parcel;
import android.os.Parcelable;

public class SignUpBody implements Parcelable {

    public String email;
    public String password;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.email);
        dest.writeString(this.password);
    }

    public SignUpBody() {
    }

    protected SignUpBody(Parcel in) {
        this.email = in.readString();
        this.password = in.readString();
    }

    public static final Creator<SignUpBody> CREATOR = new Creator<SignUpBody>() {
        @Override
        public SignUpBody createFromParcel(Parcel source) {
            return new SignUpBody(source);
        }

        @Override
        public SignUpBody[] newArray(int size) {
            return new SignUpBody[size];
        }
    };
}

File: app/src/main/java/com/travlog/android/apps/services/apirequests/SignUpBody.kt
package com.travlog.android.apps.services.apirequests

import android.os.Parcel
import android.os.Parcelable

class SignUpBody() : Parcelable {

    var email = ""
    var password = ""

    constructor(parcel: Parcel) : this() {
        email = parcel.readString()
        password = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(email)
        parcel.writeString(password)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SignUpBody> {
        override fun createFromParcel(parcel: Parcel): SignUpBody {
            return SignUpBody(parcel)
        }

        override fun newArray(size: Int): Array<SignUpBody?> {
            return arrayOfNulls(size)
        }
    }

    override fun toString(): String {
        return "SignUpBody(email='$email', password='$password')"
    }
}
